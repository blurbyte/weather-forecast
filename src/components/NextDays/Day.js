import React from 'react';
import styled from 'styled-components';

import Text from '../Text';
import WeatherIcon from '../WeatherIcon';

const Wrapper = styled.li`
  margin-top: 2rem;
`;

const Info = styled.div`
  display: flex;
  margin: 0.4rem 0;
`;

const Temperatures = styled.div`
  margin-left: 2rem;
  margin-top: 0.4rem;
`;

const Temperature = styled(Text).attrs({
  big: true
})`
  line-height: 1.2;
`;

const Day = () => (
  <Wrapper>
    <Text big>Thursday</Text>
    <Info>
      <WeatherIcon small />
      <Temperatures>
        <Temperature>20</Temperature>
        <Temperature dimmed>14</Temperature>
      </Temperatures>
    </Info>
    <Text>Clear sky</Text>
  </Wrapper>
);

export default Day;
