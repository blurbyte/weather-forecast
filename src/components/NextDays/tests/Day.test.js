import React from 'react';
import renderer from 'react-test-renderer';

import Day from '../Day';

jest.mock('../../WeatherIcon', () => 'WeatherIcon');

test('renders correctly', () => {
  const props = {
    weatherId: 800,
    description: 'clear sky',
    tempMax: 18.75,
    tempMin: 9.7,
    timestamp: '5/21/2018'
  };

  const component = renderer.create(<Day {...props} />);
  expect(component.toJSON()).toMatchSnapshot();
});
