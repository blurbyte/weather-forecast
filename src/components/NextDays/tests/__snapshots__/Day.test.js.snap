// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c1 {
  font-size: 1.8rem;
  color: #222222;
  margin: 0;
  text-transform: uppercase;
}

.c5 {
  font-size: 1.8rem;
  color: rgba(34,34,34,0.5);
  margin: 0;
  text-transform: uppercase;
}

.c6 {
  font-size: 1.4rem;
  color: #222222;
  margin: 0;
  text-transform: none;
}

.c0 {
  margin-top: 2rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0.4rem 0;
}

.c3 {
  margin-left: 2rem;
  margin-top: 0.4rem;
}

.c4 {
  line-height: 1.2;
}

<li
  className="c0"
>
  <p
    className="c1"
  >
    Thursday
  </p>
  <div
    className="c2"
  >
    <WeatherIcon
      small={true}
    />
    <div
      className="c3"
    >
      <p
        className="c4 c1"
      >
        20
      </p>
      <p
        className="c4 c5"
      >
        14
      </p>
    </div>
  </div>
  <p
    className="c6"
  >
    Clear sky
  </p>
</li>
`;
